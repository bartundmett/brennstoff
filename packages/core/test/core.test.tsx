import * as React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import {
  Brennstoff,
  sharedStyles,
  variantStyles,
  styled,
  componentStyles,
} from '../src';

const testTheme = {
  components: {
    test: {
      shared: {
        color: 'white',
      },
      size: {
        s: {
          fontSize: '13px',
        },
        m: {
          fontSize: '14px',
        },
      },
      variant: {
        primary: {
          backgroundColor: 'red',
        },
        secondary: {
          backgroundColor: 'green',
        },
      },
    },
  },
};

describe('@brennstoff/basetheme', () => {
  it('works', () => {
    expect(true).toBe(true);
  });
});

describe('sharedStyles helper', () => {
  it('renders shared styles from theme', () => {
    const shared = sharedStyles('test');
    expect(
      shared({
        theme: testTheme,
      }),
    ).toEqual({
      color: 'white',
    });
  });
});

describe('variants helper', () => {
  it('forwards themed variants to component styles', () => {
    const variants = variantStyles('test', 'size');
    const Component = styled.div<{ size: 's' | 'm' }>`
      ${variants}
    `;
    const { container, rerender } = render(
      <Brennstoff theme={testTheme}>
        <Component size="m">Test</Component>
      </Brennstoff>,
    );
    expect(container.firstChild).toHaveStyleRule('font-size', '14px');

    rerender(
      <Brennstoff theme={testTheme}>
        <Component size="s">Test</Component>
      </Brennstoff>,
    );

    expect(container.firstChild).toHaveStyleRule('font-size', '13px');
  });
});

describe('componentStyles helper', () => {
  it('combines shared and all variants', () => {
    const styles = componentStyles('test', ['size', 'variant']);
    const Component = styled.div<{
      size: 's' | 'm';
      variant: 'primary' | 'secondary';
    }>`
      ${styles}
    `;
    const { container, rerender } = render(
      <Brennstoff theme={testTheme}>
        <Component size="m" variant="primary">
          Test
        </Component>
      </Brennstoff>,
    );
    expect(container.firstChild).toHaveStyleRule('font-size', '14px');

    rerender(
      <Brennstoff theme={testTheme}>
        <Component size="s" variant="secondary">
          Test
        </Component>
      </Brennstoff>,
    );

    expect(container.firstChild).toHaveStyleRule('font-size', '13px');
  });
});

describe('themeprovider', () => {
  it('renders with provider', () => {
    const { container } = render(<Brennstoff theme={{}}>test</Brennstoff>);
    expect(container).toMatchSnapshot();
  });
});
