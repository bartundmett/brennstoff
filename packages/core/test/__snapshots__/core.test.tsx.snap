// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themeprovider renders with provider 1`] = `
<div>
  test
</div>
`;
